core/dbt/graph/thread_pool.py

```diff
@@ -0,0 +1,18 @@
+from __future__ import annotations
+
+from multiprocessing.pool import ThreadPool
+
+
+class DbtThreadPool(ThreadPool):
```

Comment:
We created `DbtThreadPool` so that we can have visibility on whether `.close()` has been called on the pool. This class is now used instead of `ThreadPool`.